src/Promitor.Core.Scraping/Configuration/Model/Metrics/ResourceTypes/ContainerInstanceMetricDefinition.cs

```diff
@@ -1,7 +1,19 @@
-﻿namespace Promitor.Core.Scraping.Configuration.Model.Metrics.ResourceTypes
+﻿using System.Collections.Generic;
+
+namespace Promitor.Core.Scraping.Configuration.Model.Metrics.ResourceTypes
 {
     public class ContainerInstanceMetricDefinition : MetricDefinition
     {
+        public ContainerInstanceMetricDefinition()
+        {
+        }
+
+        public ContainerInstanceMetricDefinition(AzureMetricConfiguration azureMetricConfiguration, string description, string name, string resourceGroupName, string containerGroup, Dictionary<string, string> labels, Scraping scraping)
+            : base(azureMetricConfiguration, description, name, resourceGroupName, labels, scraping)
+        {
```

Comment:
Same question for all properties - Should we enforce valid configurations via guarding or not?

Reply:
I think it makes sense so I'll update that. The slight catch here is that I think AutoMapper will be using the zero argument constructors until we remove them (assuming we can). But I can add the guards in here, and when we get rid of the zero argument constructors it'll be enforced correctly.

Reply:
Perfect, thanks!

Reply:
@tomkerkhove slight hiccup with adding the Guards in. The problem is that the validation steps rely on being able to create an invalid object, and there are then validation rules to ensure all the required properties have been set.

For example, `AzureStorageQueueMetricsDeclarationValidationStepTests.AzureStorageQueuesMetricsDeclaration_DeclarationWithoutAccountName_Fails` will fail if I guard against the account name being null or empty because it tries to create a `StorageQueueMetricDefinition` with an empty account name.

So I think if we wanted to go down that route we'd need to rethink how the validation works.

What's your thoughts on that?

Reply:
I think it's fine to leave out the guards for now then, it's clear that it's covered for now.

Once we've releases 1.0 we can revise a few things like this

Reply:
Thanks btw!

Resolving this conversation as it's no longer applicable.